src/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,2689 @@
+*(a::MPolyElem_dec, b::SubQuoElem) = SubQuoElem(a*b.coeffs, b.parent)
+*(a::MPolyElem, b::SubQuoElem) = SubQuoElem(a*b.coeffs, b.parent)
+*(a::Int, b::SubQuoElem) = SubQuoElem(a*b.coeffs, b.parent)
+*(a::Integer, b::SubQuoElem) = SubQuoElem(a*b.coeffs, b.parent)
+*(a::fmpq, b::SubQuoElem) = SubQuoElem(a*b.coeffs, b.parent)
+==(a::SubQuoElem, b::SubQuoElem) = iszero(a-b)
+
+function sub(F::FreeMod, O::Array{<:FreeModuleElem, 1}, task::Symbol = :none)
+  s = SubQuo(F, O)
+  if task == :none || task == :module
+    return s
+  else
+    emb = hom(s, F, O)
+    task == :store && register_morphism!(emb)
+    task == :morphism && return emb
+    return s, emb
+  end
+end
+
+function sub(F::FreeMod, O::Array{<:SubQuoElem, 1}, task::Symbol = :none)
+  s = SubQuo(F, [x.repres for x = O])
+  return sub(F, s, task)
+  #=if task == :none
+    return s
+  else
+    emb = hom(s, F, [x.repres for x = O])
+  end=#
+end
+
+function sub(F::FreeMod, s::SubQuo, task::Symbol = :none)
+  @assert !isdefined(s, :quo)
+  @assert s.F == F
+  if task == :none || task == :module
+    return s
+  else
+    emb = hom(s, F, [FreeModuleElem(x.repres.coords, F) for x in gens(s)])
+    task == :store && register_morphism!(emb)
+    task == :morphism && return emb 
+    return s, emb
+  end
+end
+
+function sub(S::SubQuo, O::Array{<:SubQuoElem, 1}, task::Symbol = :none, check_only_relations = false)
+  if !check_only_relations
+    @assert all(x -> x.parent === S, O)
+  else
+    if length(O) > 0
+      @assert O[1].parent.quo == S.quo
+    end
+  end
+  t = SubQuo(S.F, [x.repres for x in O])
+  if isdefined(S, :quo)
+    t.quo = S.quo
+    t.sum = sum(t.sub, t.quo)
+  end
+  if task == :none || task == :module
+    return t
+  else
+    emb = hom(t, S, O)
+    task == :store && register_morphism!(emb)
+    task == :morphism && return emb 
+    return t, emb
+  end
+
+  #=t = sub(S.F, O, task)
+  if task != :none
+    t,emb = t
+  end
+  if isdefined(S, :quo)
+    s = quo(t, collect(S.quo.gens))
+    if task == :none
+      return s
+    else
+      emb2 = hom(s, S, [SubQuoElem(x.repres, S) for x in gens(s)])
+    end
+  else
+    return t
+  end=#
+end
+
+function quo(F::FreeMod, O::Array{<:FreeModuleElem, 1}, task::Symbol = :none)
+  S = SubQuo(F, basis(F))
+  Q = SubQuo(S, O)
+
+  return return_quo_wrt_task(F, Q, task)
+end
+
+function quo(F::FreeMod, O::Array{<:SubQuoElem, 1}, task::Symbol = :none)
+  S = SubQuo(F, basis(F))
+  Q = SubQuo(S, [x.repres for x = O])
+  
+  return return_quo_wrt_task(F, Q, task)
+end
+
+function quo(F::SubQuo, O::Array{<:FreeModuleElem, 1}, task::Symbol = :none)
+  if length(O) > 0
+    @assert parent(O[1]) == F.F
+  end
+  if isdefined(F, :quo)
+    #F.sub[Val(:S), 1]
+    #[F.quo.gens[Val(:O), i] for i = 1:length(F.quo.gens.O)] 
+    oscar_assure(F.quo.gens)
+    s = Singular.smodule{elem_type(base_ring(F.quo.gens.SF))}(base_ring(F.quo.gens.SF), [convert(F.quo.gens.SF, x) for x = [O; F.quo.gens.O]]...)
```

Comment:
We should (and will) improve Singular.jl to allow writing this instead:
```suggestion
    s = Singular.smodule(base_ring(F.quo.gens.SF), [convert(F.quo.gens.SF, x) for x = [O; F.quo.gens.O]]...)
```